app/src/main/java/com/example/tasky/agenda/di/AgendaModule.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.tasky.agenda.di
+
+import com.example.tasky.agenda.data.DateGeneratorImpl
+import com.example.tasky.agenda.data.remote.AgendaRepositoryImpl
+import com.example.tasky.agenda.data.remote.AgendaService
+import com.example.tasky.agenda.domain.AgendaRepository
+import com.example.tasky.agenda.domain.DateGenerator
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import retrofit2.Retrofit
+
+
+@Module
+@InstallIn(SingletonComponent::class)
+class AgendaModule {
+
+    @Provides
+    fun providesAgendaRepository(retrofit: Retrofit): AgendaService {
+        return retrofit.create(AgendaService::class.java)
+    }
+}
+
+@Module
+@InstallIn(SingletonComponent::class)
+sealed class AgendaBindsModule {
+
+    @Binds
+    abstract fun bindMonthGenerator(monthGeneratorImpl: DateGeneratorImpl): DateGenerator
+
+    @Binds
```

Comment:
Why do you sometimes use Binds, and other times Provides?

Reply:
I use Provides when I have to manually instantiate the class in the method block in the module for example a retrofit service. And Binds when I only have to link the Implementation to the interface, so when the viewModel ask for a DateGenerator, hilt knows that it need to instantiate a DateGeneratorImpl and pass it. Since most of the times my Impl uses params in the constructor provided by hilt it doesnt require a Provides only a Binds

Reply:
Gotcha, sounds good